packages/xstate-wallet/src/chain.ts

```diff
@@ -235,7 +235,7 @@ const chainLogger = logger.child({module: 'chain'});
 export class ChainWatcher implements Chain {
   private _adjudicator?: Contract;
   private _assetHolders: Contract[];
-  private mySelectedAddress: string | null;
+  private mySelectedAddress: string | null = (window.ethereum || {}).selectedAddress;
```

Comment:
```suggestion
  private mySelectedAddress: string | null = window.ethereum?.selectedAddress ?? null;
```

I think your code change would be a runtime `TypeError`  when `window.ethereum` is falsey.

Typescript does not catch this because `window.ethereum` has type `any`, therefore so does `window.ethereum || {}`.